#8 add simple maven cucumber example

joergsesterhenn opened the pull request:
This is one of my first PRs so please let me know if anything needs to be changed 😉

joergsesterhenn committed:
#8 add simple maven cucumber example

haroon-sheikh commented on examples/maven/pom.xml:
```diff
@@ -0,0 +1,75 @@
+			<version>${extentreports.version}</version>
+			<scope>provided</scope>
+		</dependency>
+	</dependencies>
```
Can we remove any dependencies that are not being used?

haroon-sheikh commented on examples/maven/pom.xml:
```diff
@@ -0,0 +1,75 @@
+		<junit.version>4.12</junit.version>
+		<maven-compiler-plugin.version>3.7.0</maven-compiler-plugin.version>
+		<maven-source-plugin.version>3.0.1</maven-source-plugin.version>
+		<maven-javadoc-plugin.version>2.10.4</maven-javadoc-plugin.version>
```
Remove unused properties

haroon-sheikh commented on examples/maven/pom.xml:
```diff
@@ -0,0 +1,75 @@
+    <dependency>
+      <groupId>com.sitture</groupId>
+      <artifactId>cucumber-jvm-extentreport</artifactId>
+      <version>2.0.3-SNAPSHOT</version>
```
lets move this version into properties?

haroon-sheikh reviewed (changes requested):
Hi @joergsesterhenn, thanks a ton for your contribution. Just a few minor changes to make please.

joergsesterhenn committed:
#8 cleaning up maven example

- removed declared transitive dependencies
- removed not needed properties
- added property for dependency version
- removed config for encoding in compiler-plugin 
  since it defaults to project.source.encoding
- moved output to target so that it gets deleted 
  when mvn clean is called

joergsesterhenn commented:
I hope this is acceptable?
